Src/DependencyCollector/NetCore.Tests/DependencyTrackingTelemetryModuleTestNetCore.cs

```diff
@@ -0,0 +1,169 @@
+            }
+        }
+
+        /// <summary>
+        /// Tests that dependency is collected properly when there is no parent activity.
+        /// </summary>
+        [TestMethod]
+        [Timeout(5000)]
+        public async Task TestDependencyCollectionNoParentActivity()
+        {
+            ITelemetry sentTelemetry = null;
+
+            var channel = new StubTelemetryChannel
+            {
+                OnSend = telemetry =>
+                {
+                    // The correlation id lookup service also makes http call, just make sure we skip that
+                    DependencyTelemetry depTelemetry = telemetry as DependencyTelemetry;
+                    if (depTelemetry != null &&
+                        !depTelemetry.Data.StartsWith(FakeProfileApiEndpoint, StringComparison.OrdinalIgnoreCase))
+                    {
+                        Assert.IsNull(sentTelemetry);
+                        sentTelemetry = telemetry;
+                    }
+                },
+                EndpointAddress = FakeProfileApiEndpoint
+            };
+
+            var config = new TelemetryConfiguration
+            {
+                InstrumentationKey = IKey,
+                TelemetryChannel = channel
+            };
+
+            config.TelemetryInitializers.Add(new OperationCorrelationTelemetryInitializer());
+
+            using (var module = new DependencyTrackingTelemetryModule())
+            {
+                module.ProfileQueryEndpoint = FakeProfileApiEndpoint;
+                module.Initialize(config);
+
+                var url = new Uri("http://bing.com");
+
+                var request = new HttpRequestMessage(HttpMethod.Get, url);
+                await new HttpClient().SendAsync(request);
+
+                // on netcoreapp1.0 DiagnosticSource event is fired asycronously, let's wait for it 
+                Assert.IsTrue(SpinWait.SpinUntil(() => sentTelemetry != null, TimeSpan.FromSeconds(1)));
+
+                var item = (DependencyTelemetry)sentTelemetry;
+                Assert.AreEqual(url, item.Data);
+                Assert.AreEqual(url.Host, item.Target);
+                Assert.AreEqual("GET " + url.AbsolutePath, item.Name);
+                Assert.IsTrue(item.Duration > TimeSpan.FromMilliseconds(0), "Duration has to be positive");
+                Assert.AreEqual(RemoteDependencyConstants.HTTP, item.Type, "HttpAny has to be dependency kind as it includes http and azure calls");
+                Assert.IsTrue(
+                    DateTime.UtcNow.Subtract(item.Timestamp.UtcDateTime).TotalMilliseconds <
+                    TimeSpan.FromMinutes(1).TotalMilliseconds,
+                    "timestamp < now");
+                Assert.IsTrue(
+                    item.Timestamp.Subtract(DateTime.UtcNow).TotalMilliseconds >
+                    -TimeSpan.FromMinutes(1).TotalMilliseconds,
+                    "now - 1 min < timestamp");
+                Assert.AreEqual("200", item.ResultCode);
```

Comment:
Minor: Here we assume that bing.com does not fail, should we use local endpoint instead if we'd like to check the status code or should we skip status code validation here? It might've been the same in the other tests of ours already, though... but we can start a good habit here, instead :)